### PR TITLE
Fix libspdm_x509_get_extension_data description and logic error

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #2590

### CODE REVIEW

File: os_stub/cryptlib_openssl/pk/x509.c [hunk 1]
```diff
@@ -1414,6 +1410,14 @@ bool libspdm_x509_get_extension_data(const uint8_t *cert, size_t cert_size,
         /* reset to 0 if not found */
         oct_length = 0;
     }
+
+    /* the and the cert extension is found, but the oid extension is not found; */
+    if (i == sk_X509_EXTENSION_num(extensions)) {
```
Comment (jyao1) [inline os_stub/cryptlib_openssl/pk/x509.c:1415]: Is this same as (status == false)?

File: include/hal/library/cryptlib/cryptlib_cert.h [hunk 2]
```diff
@@ -112,8 +112,11 @@ extern bool libspdm_x509_get_issuer_name(const uint8_t *cert, size_t cert_size,
  * @param[out]     extension_data       Extension bytes.
  * @param[in, out] extension_data_size  Extension bytes size.
  *
- * @retval  true
- * @retval  false
+ * @return true   If the returned extension_data_size == 0, it means that cert and oid are valid, but the oid extension is not found;
+ *                If the returned extension_data_size != 0, it means that cert and oid are valid, and the oid extension is found;
+ * @return false  If the returned extension_data_size == 0, it means that cert or oid are invalid;
+ *                If the returned extension_data_size != 0, it means that cert and oid are valid, and the oid extension is found,
```
Comment (steven-bellock) [inline include/hal/library/cryptlib/cryptlib_cert.h:118]: In this case does `extension_data_size` return the size of the `extension_data`?

File: os_stub/cryptlib_openssl/pk/x509.c (hunk 1 above)
Comment (Wenxing-hou) [reply os_stub/cryptlib_openssl/pk/x509.c]: Thanks. I have changed the code.

File: include/hal/library/cryptlib/cryptlib_cert.h (hunk 2 above)
Comment (Wenxing-hou) [reply include/hal/library/cryptlib/cryptlib_cert.h:118]: Yes. The returned `extension_data_size` is the size of the `extension_data`. Thanks.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**